staging/src/k8s.io/component-base/logs/logs.go

```diff
@@ -29,16 +29,25 @@ import (
 
 const logFlushFreqFlagName = "log-flush-frequency"
 
-var logFlushFreq = pflag.Duration(logFlushFreqFlagName, 5*time.Second, "Maximum number of seconds between log flushes")
+var (
+	klogFlags    flag.FlagSet
+	packageFlags pflag.FlagSet
```

Comment:
What is the advantage? Just avoiding the direct dependency, or also making the package usable with a Go `flag.FlagSet`? The latter would be useful, but doesn't seem to be straightforward because Go FlagSet doesn't implement this interface.

In the [follow-up PR for deprecating klog flags](https://github.com/kubernetes/kubernetes/pull/105042/commits) I am depending on `pflag.Flag.Deprecated`. That would be impossible without the direct dependency.

Reply:
I'm adding `func AddGoFlags(fs *flag.FlagSet)`. With the klog flags not being added to `flag.CommandLine` anymore automatically, test programs had to call `klog.InitFlags(nil)` directly to get those flags back - that's going in the wrong direction when considering deprecation of those flags.

Reply:
My thinking was simply to avoid the coupling.  It felt weird and wrong to have one `flag.FlagSet` and one `pflag.Flagset`.  In other code I deal with, I regularly screw up flag/pflag so I am mostly looking for "protecting the code from future me" in this comment